src/Esprima/Jsx/JsxToken.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Esprima.Jsx
+{
+    public enum JsxTokenType
+    {
+        JsxIdentifier,
+        JsxText
+    };
+
+    public sealed class JsxToken : Token
```

Comment:
this might not fly well with my attempt of changing Token to struct... but I think I can work around it struct-based approach seems beneficial

Reply:
That also seems beneficial to me if you can pass `Token` around without copying (i.e. by-ref).

As for the problem of inheritance, you may add an `object` field named something like `ExtensionData` to `Token`. Then `JsxParser` could store the JSX token type in that field. This info is accessed only [at one place in `JsxParser`] (https://github.com/sebastienros/esprima-dotnet/pull/258/commits/ae86c4e53e349ddac2cb9889c8eab45db84544b0#diff-3369e40bf400ba131a958de181b5592a878f88e30027aa3a1a8c2dae81e5c042R636) anyway.